### PR TITLE
[smoke-test] Add option to have use --release mode for nodes, and enable it in CI

### PR DESCRIPTION
**Pull request by igor-aptos**
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3074)

### CODE REVIEW

**Comment by igor-aptos** [comment]
rosetta::test_block is consistently failing with this. Ignoring that test for now, while we asynchronously debug that.

File: execution/executor/src/components/chunk_output.rs [hunk 1]
```diff
@@ -33,8 +33,14 @@ impl ChunkOutput {
         transactions: Vec<Transaction>,
         state_view: CachedStateView,
     ) -> Result<Self> {
+        let start = Instant::now();
         let transaction_outputs = V::execute_block(transactions.clone(), &state_view)?;
 
+        debug!(
```
Comment (zekun000) [inline execution/executor/src/components/chunk_output.rs:39]: remove?

File: .github/workflows/lint-test.yaml [hunk 2]
```diff
@@ -167,8 +167,9 @@ jobs:
       - uses: taiki-e/install-action@v1.5.6
         with:
           tool: nextest
+      # prebuild node binary, so that tests don't start before node is built.
       # --test-threads is intentionally set to reduce resource contention in ci jobs. Increasing this, increases job failures and retries.
-      - run: cargo nextest run --profile ci --package smoke-test --test-threads 6 --retries 3
+      - run: cargo build --bin=aptos-node --features=failpoints --release && LOCAL_SWARM_NODE_RELEASE=1 cargo nextest run --release --profile ci --package smoke-test --test-threads 6 --retries 3
```
Comment (zekun000) [inline .github/workflows/lint-test.yaml:172]: we probably can split the build into a separate step?

File: execution/executor/src/components/chunk_output.rs (hunk 1 above)
Comment (igor-aptos) [reply execution/executor/src/components/chunk_output.rs:39]: I was planning to leave it in, but I can remove it

File: .github/workflows/lint-test.yaml (hunk 2 above)
Comment (igor-aptos) [reply .github/workflows/lint-test.yaml:172]: can it be a separate step? How do I make sure it is running on the same machine, and output of that step is there when I start this step?

Comment (igor-aptos) [reply .github/workflows/lint-test.yaml:172]: plus if someone copies the command to re-run smoke tests locally, they are going to have a better experience with running the full command

Comment (zekun000) [reply .github/workflows/lint-test.yaml:172]: I think all steps here are run on the same machine, just need another line of -run? but not sure

**Review by zekun000 — APPROVED** [review]

**Comment by igor-aptos** [comment]
ah you meant within the same command . well that is visible the same way in the UI, and not retriable individually.

so , given the better experience to copy the whole command, I am going to leave it as is.

**Pull request merged**